lib/onigumo.ex

```diff
@@ -9,11 +9,19 @@ defmodule Onigumo do
     HTTPoison.start()
     http = http_client()
 
-    load_urls(@input_filename)
-    |> Enum.map(&download(http, &1))
+    download(http)
   end
 
-  def download(http_client, url) do
+  def download(http_client) do
+    urls = load_urls(@input_filename)
+    download(http_client, urls)
+  end
+
+  def download(http_client, urls) when is_list(urls) do
+    Enum.map(urls, &download(http_client, &1))
+  end
+
+  def download(http_client, url) when is_bitstring(url) do
```

Comment:
So if I've understood well, now we can call download as `download/1` or `download/2`. Both of cases are valid.

But I am little bit confused abotu usage. It seems that `download/2` is using only in tests?

Reply:
For now it is. I don’t see that however as a problem. The `download/2` function provides a sane API to download a URL or a list of URLs rather than an implementation detail. Thus it seems ok to me to have this function defined as public. Moreover private functions are not accessible by tests.

If there should be no functions only used by tests, then it would not be possible to split any function to smaller, atomic chunks. Even #19 suggests doing so. The function is actually a result of such split, so it is testable as a unit and not only by an integration test.